runtime/src/bank.rs

```diff
@@ -5034,6 +5034,29 @@ impl Bank {
         timings: &mut ExecuteTimings,
         log_messages_bytes_limit: Option<usize>,
     ) -> (Vec<TransactionCommitResult>, TransactionBalancesSet) {
+        self.load_execute_and_commit_transactions_with_pre_commit_callback(
+            batch,
+            max_age,
+            collect_balances,
+            recording_config,
+            timings,
+            log_messages_bytes_limit,
+            None::<fn() -> bool>,
+        )
+        .unwrap()
+    }
+
+    #[must_use]
+    pub fn load_execute_and_commit_transactions_with_pre_commit_callback(
+        &self,
+        batch: &TransactionBatch<impl TransactionWithMeta>,
+        max_age: usize,
+        collect_balances: bool,
+        recording_config: ExecutionRecordingConfig,
+        timings: &mut ExecuteTimings,
+        log_messages_bytes_limit: Option<usize>,
+        pre_commit_callback: Option<impl FnOnce() -> bool>,
```

Comment:
is it simpler to just not make this optional?
just have `impl FnOnce() -> bool`. Likely compiler will optimize the simple case of `|| true` out completely, and let's us simplify the code by not having conditional calls.

Reply:
hehe, i made them `Option`-ed by purpose. 36f853714c638b9f732b3094ef20de5b2235fa38

indeed, `|| true` will completely optimized out. but I prefer to retain `Option` here, considering this is very security sensitive code for extra safety to ensure block-verification code-path isn't affected.